### ISSUE TITLE
WrapDocument called too much on solution load

### ISSUE THREAD

**Issue by KirillOsenkov** [mono/monodevelop#6772]
Place a breakpoint in `Workbench.cs` line 876 (`WrapDocument()`) during solution load with an open file.

It will get called a lot, just to get the active project. It seems that we should be caching the active project and document in the `MonoDevelopDocumentTrackingService.cs`, instead of retrieving it every time via FindDocument.

```

  | MonoDevelop.Ide.Gui.Workbench.WrapDocument Line 876 |  
-- | -- | --
  | MonoDevelop.Ide.Gui.Workbench.get_ActiveDocument Line 172 |  
  | MonoDevelop.Ide.RoslynServices.MonoDevelopDocumentTrackingService.GetActiveDocument Line 96 |  
  | Microsoft.CodeAnalysis.SolutionCrawler.SolutionCrawlerRegistrationService+WorkCoordinator+IncrementalAnalyzerProcessor.GetActiveProject Line 267 |  
  | Microsoft.CodeAnalysis.SolutionCrawler.SolutionCrawlerRegistrationService+WorkCoordinator+IncrementalAnalyzerProcessor+LowPriorityProcessor+<ExecuteAsync>d__3.MoveNext Line 53 |  


```

> VS bug [#753459](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/753459)

**Comment by KirillOsenkov** [comment]
Looks like it's called 13 times during a regular file open.

**Comment by Therzok** [comment]
https://github.com/mono/monodevelop/pull/6855

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-9] Comment by **xamarinc**



Fixed in version 8.0.0.1126 (master)



Author: Marius Ungureanu  
 Commit: 65e918570a43773e189ef12f1f657204ffdd4642 (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-11] Comment by **xamarinc**



Fixed in version 8.0.0.1154 (release-8.0)



Author: Marius Ungureanu  
 Commit: 65e918570a43773e189ef12f1f657204ffdd4642 (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-12] Comment by **xamarinc**



Fixed in version 8.0.0.1738 (release-8.0)



Author: Marius Ungureanu  
 Commit: 65e918570a43773e189ef12f1f657204ffdd4642 (xamarin/monodevelop)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-18] Comment by **xamarinc**



Fixed in version 8.6.0.802 (master)



Author: Marius Ungureanu  
 Commit: 65e918570a43773e189ef12f1f657204ffdd4642 (xamarin/monodevelop)
